src/Examine/Search/IQueryExecutor.cs

```diff
@@ -12,7 +12,21 @@ public interface IQueryExecutor
         /// <returns></returns>
         ISearchResults Execute(int maxResults = 500);
 
-        
 
+        /// <summary>
+        /// Executes the query with skip already applied. Use when you only need a single page of results to save on memory. To obtain additional pages you will need to execute the query again.
+        /// </summary>
+        /// <param name="skip">Number of results to skip</param>
+        /// <param name="take">Number of results to take</param>
+        /// <returns></returns>
+        ISearchResults ExecuteWithSkip(int skip, int? take = null);
```

Comment:
This will have to unfortunately also use interesting work arounds since we can't modify the interfaces for this version. It could be part of another interface like IQueryExecutorWithSkip or it could be an interface that is named in order to replace this one in future versions like IQueryExecutor2 that inherits from this one. Totally up for other ideas if you have them.

Reply:
Implemented as suggested.